hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/StandaloneEncryptedFileSystem.java

```diff
@@ -135,7 +137,11 @@ public boolean exists(Path path) throws IOException {
     public boolean delete(Path path, boolean recursive) throws IOException {
         // EncryptedFileSystem does not handle recursive deletes because it needs to rely on the KeyStorageStrategy.
         // Here we know we are using the FileKeyStorageStrategy and can therefore delegate deletes to the FileSystem
-        // which will remove both the data and key material files.
+        // which will remove both the data and key material files. If the path is a directory then the recursive delete
+        // will handle removing the key material files, otherwise we will explicitly delete the key material here.
+        if (fs.isFile(path)) {
+            keyStore.remove(path.toString());
```

Comment:
does the order of the two delete calls matter? in theory, if you fail the `delegate.delete` call then `exists()` returns true for the file, but it still can't be read. maybe doesn't matter.

Reply:
don't think it matters either way? think this ends up reading a bit more nicely though since I don't ned to keep around the result of the `delete.delete` call

Reply:
works for me